### DISCUSSION TITLE
fail to compile meep becaause of   --with-mpi

### DISCUSSION THREAD

**Discussion by longlong10086** [NanoComp/meep#2031] [Q&A]
I  tried to install meep1.22.0 with source code ,and I got an error .

- error
checking for mpic++... /opt/meep/meep.deps/bin/mpicc++
checking for MPI_Init... no
checking for MPI_Init in -lmpi... no
checking for MPI_Init in -lmpich... no
configure: error: could not find mpi library for --with-mpi

- system
centos7.9  ,3.10.0-1160.21.1.el7.x86_64

- mpi
/opt/meep/meep.deps/lib/libmpi.so
/usr/lib64/mpich/lib/libmpich.so

- command
RPATH_FLAGS="-Wl,-rpath,/opt/meep/meep.deps/lib:/opt/meep/meep.deps/lib/openmpi"
MY_LDFLAGS="-L/opt/meep/meep.deps/lib -L/opt/meep/meep.deps/lib/openmpi ${RPATH_FLAGS}"
MY_CPPFLAGS="-I/opt/meep/meep.deps/include -I/opt/meep/meep.deps/include/openmpi"

export PATH=/opt/meep/meep.deps/bin:$PATH
export LD_LIBRARY_PATH=/usr/lib64/mpich/lib:/opt/meep/meep.deps/lib:$LD_LIBRARY_PATH
export LIBRARY_PATH=/usr/lib64/mpich/lib:/opt/meep/meep.deps/lib:$LIBRARY_PATH
#export LDFLAGS="-L/usr/lib64/mpich/lib -L/opt/meep/meep.deps/lib -L/opt/meep/meep.deps/lib/openmpi  -lmpi -lmpich"

cd meep/
sh autogen.sh   --enable-shared -  -prefix=/opt/meep/meep      --with-mpi     --with-openmp        PYTHON=python3 MPICC=/opt/meep/meep.deps/bin/mpicc         MPICXX=/opt/meep/meep.deps/bin/mpicc++                                              
LDFLAGS="-L/usr/lib64/mpich/lib  ${MY_LDFLAGS}" CPPFLAGS="${MY_CPPFLAGS}" 
--with-libctl=/opt/meep/meep.deps/share/libctl

How to fix the problem.Hope to get your help.

**Comment by stevengj** [comment]
Your MPI installation is broken if `mpic++` doesn't automatically link a library with `MPI_Init`.

You can look in the `config.log` file (search for `MPI_Init`) to find the exact compiler error message.

**Comment by pavlo2014** [comment]
Is mpicc++ stored in /opt/meep/meep.deps/bin/ ? It is quite unusual location. 

This is how variables are defined in my file bashrc file (currently being used on an Ununtu computer but previosly used on a Centos 7 computer too):

LDFLAGS="-L/usr/local/lib -lm" export LDFLAGS
CPPFLAGS="-I/usr/local/include" export CPPFLAGS
PATH="$PATH:/usr/local/bin:/home/user/.local/bin" export PATH
PYTHONPATH="$PYTHONPATH:/usr/local/lib/python3.8/dist-packages:/usr/local/lib/python3.8/site-packages" export PYTHONPATH

LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/lib:/usr/local/lib64 export LD_LIBRARY_PATH
PKG_CONFIG_PATH=$PKG_CONFIG_PATH:/usr/local/lib/pkgconfig export PKG_CONFIG_PATH
HDF5_MPI="ON" export HDF5_MPI
